ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/events/WitnessesTable.scala

```diff
@@ -3,23 +3,24 @@
 
 package com.daml.platform.store.dao.events
 
-import anorm.{BatchSql, NamedParameter}
-import com.daml.lf.data.Ref.LedgerString
+import anorm.{BatchSql, NamedParameter, ToStatement}
+import com.daml.ledger.EventId
 import com.daml.platform.store.Conversions._
+import com.daml.platform.store.DbType
 
 /**
   * A table storing a flattened representation of a [[WitnessRelation]]
   */
-private[events] sealed abstract class WitnessesTable(
-    tableName: String,
-    idColumn: String,
-    witnessColumn: String,
+private[events] sealed abstract class WitnessesTable[Id: ToStatement](
+    protected val tableName: String,
+    protected val idColumn: String,
+    protected val witnessColumn: String,
 ) {
 
-  private val insert =
+  protected val insert =
```

Comment:
Since this query needs to be overridden for contracts anyway, maybe make it abstract here and move this query to the `EventWitnessTable` implementation?

Reply:
Makes sense, thanks, this is a leftover from the original implementation.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/5541/commits/8f9aa6886333a0ebe9866428861a73a03a7efdad